### PR TITLE
OBSDOCS-400: update-CLI-command-for-getting-federation-host-for-monitoring

### PR DESCRIPTION
**Pull request by bburt-rh**
Version(s): 4.12+

Issue: https://issues.redhat.com/browse/OBSDOCS-400

Link to docs preview: https://73192--ocpdocs-pr.netlify.app/openshift-enterprise/latest/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-third-party-monitoring-apis

QE review:
- [x] QE has approved this change.

Additional information:
This PR updates the [querying metrics using the federation endpoint procedure](https://docs.openshift.com/container-platform/4.15/monitoring/accessing-third-party-monitoring-apis.html#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-third-party-monitoring-apis) to include a step to get the federation host and to include the host in the curl command.

### CODE REVIEW

File: modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc [hunk 1]
```diff
@@ -40,24 +39,27 @@ You can only use bearer token authentication to access the federation endpoint.
...
-. Query metrics from the `/federate` route.
-The following example queries `up` metrics:
+. Get the federation route URL by running the following command:
 +
 [source,terminal]
 ----
-$ curl -G -s -k -H "Authorization: Bearer $TOKEN" \
-    'https://<federation_host>/federate' \ <1>
-    --data-urlencode 'match[]=up'
+$ HOST=$(oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={.spec.host})
```
Comment (juzhao) [inline modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc:53]: same as I mentioned in https://github.com/openshift/openshift-docs/pull/72176#discussion_r1506909353, the `cluster-monitoring-view` is not able to view routes under openshift-monitoring

bound cluster-monitoring-view cluster role to common user testuser-11 by admin
```
$ oc adm policy add-cluster-role-to-user cluster-monitoring-view testuser-11
Warning: User 'testuser-11' not found
clusterrole.rbac.authorization.k8s.io/cluster-monitoring-view added: "testuser-11"
```
oc login with the user, failed to get thanos-querier route
```
# oc whoami
testuser-11

# HOST=$(oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={.spec.host})
Error from server (Forbidden): routes.route.openshift.io "prometheus-k8s-federate" is forbidden: User "testuser-11" cannot get resource "routes" in API group "route.openshift.io" in the namespace "openshift-monitoring"
```
we should do the same in https://docs.openshift.com/container-platform/4.15/monitoring/accessing-third-party-monitoring-apis.html#accessing-a-monitoring-web-service-api_accessing-third-party-monitoring-apis
add one prerequisite and note
prerequisite
`You are logged in to an account that has permission to get the prometheus federate route.`
note
`If your account does not have permission to get the prometheus federate route, a cluster administrator can provide the URL for the route.`

File: modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc [hunk 2]
```diff
@@ -40,24 +39,27 @@ You can only use bearer token authentication to access the federation endpoint.
 
 .Procedure
 
-. Retrieve the bearer token:
+. Retrieve the bearer token by running the following the command:
 +
 [source,terminal]
 ----
 $ TOKEN=`oc whoami -t`
```
Comment (juzhao) [inline modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc:46]:
```suggestion
$ TOKEN=$(oc whoami -t)
```
for consistency, it's better to update to it, see the same command in https://docs.openshift.com/container-platform/4.15/monitoring/accessing-third-party-monitoring-apis.html

File: modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc [hunk 3]
```diff
@@ -40,24 +39,27 @@ You can only use bearer token authentication to access the federation endpoint.
...
 ----
 
-. Query metrics from the `/federate` route.
-The following example queries `up` metrics:
+. Get the federation route URL by running the following command:
```
Comment (juzhao) [inline modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc:49]:
```suggestion
. Get the prometheus federation route URL by running the following command:
```

File: modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus.adoc:53]: Added per your suggestion

**Comment by kcarmichael08** [comment]
Nice job! LGTM.

**Pull request merged**